app/controllers/katello/api/v2/debs_controller.rb

```diff
@@ -88,9 +88,13 @@ def available_for_content_view_version(version)
     def custom_index_relation(collection)
       applicable = ::Foreman::Cast.to_bool(params[:packages_restrict_applicable]) || params[:host_id]
       upgradable = ::Foreman::Cast.to_bool(params[:packages_restrict_upgradable])
+      not_installed = ::Foreman::Cast.to_bool(params[:packages_restrict_not_installed])
 
       if upgradable
         collection = collection.installable_for_hosts(@hosts)
+      elsif not_installed && params[:host_id]
+        host = @hosts.first
+        collection = Katello::Deb.deb_installable_for_host(host)
```

Comment:
Wouldn't you want to run `deb_installable_for_host` on `Host.where(id: params[:host_id]).first` and not just `@hosts.first`?

Reply:
It is already found correctly in `find_hosts` method below. I agree that this is not straight forward from the logic point of view. I copied the logic from https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/packages_controller.rb#L61 .